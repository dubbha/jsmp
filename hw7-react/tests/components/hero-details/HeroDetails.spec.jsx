import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HeroDetails, mapStateToProps, mapDispatchToProps }
  from 'components/hero-details/HeroDetails';

jest.mock('components/heroes/heroes.actions', () => ({
  updateHero: jest.fn(),
}));

jest.mock('components/heroes/heroes.selectors', () => ({
  heroReselector: (state, props) => (
    state.heroes.filter(el => el.id === props.match.params.id)[0]
  ),
}));

describe('HeroDetails', () => {
  it('should render dashboard top hero element', () => {
    const props = {
      hero: { id: 1, name: 'Primus' },
      updateHero: jest.fn(),
      history: { goBack: jest.fn() },
      match: { url: 'test_url' },
    };
    const wrapper = shallow(<HeroDetails {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const state = {
      heroes: [
        { id: 4, name: 'Optimus Four' },
        { id: 5, name: 'Optimus Five' },
      ]
    };

    const props = { match: { params: { id: 4 } } };

    expect(mapStateToProps(state, props)).toEqual({
      hero: { id: 4, name: 'Optimus Four' },
    });
  });

  it('should map dispatch to props', () => {
    const dispatch = jest.fn();
    const hero = { id: 1, name: 'Primus' };
    expect(mapDispatchToProps(dispatch).updateHero(hero)).toEqual(dispatch(hero));
  });

  it('should call handleChange on input change', () => {
    const props = {
      hero: { id: 1, name: 'Primus' },
      updateHero: jest.fn(),
      history: { goBack: jest.fn() },
      match: { url: 'test_url' },
    };
    const wrapper = shallow(<HeroDetails {...props} />);
    wrapper.find('input').simulate('change', { target: { value: 'test_value' } });
    expect(props.updateHero).toBeCalled();
  });
});
