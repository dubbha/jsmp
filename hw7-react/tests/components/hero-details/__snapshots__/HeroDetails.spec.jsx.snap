// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetails should render dashboard top hero element 1`] = `
<div
  className="hero-details"
>
  <h2>
    Primus
     details!
  </h2>
  <div>
    <label>
      id: 
    </label>
    1
  </div>
  <div>
    <label>
      name: 
    </label>
    <input
      onChange={[Function]}
      placeholder="name"
      value="Primus"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;
